docs/cookbook/import-categories.md

```diff
@@ -0,0 +1,198 @@
+# Import categories
+
+Category import is one of the specific imports.
+Due to the tree structure it's really easy to create ineffective script.
+In this cookbook we highlight some concepts and how to improve category import performance.
+
+## Task
+
+> _As an eshop owner I want to import categories from IS, so I don't need to edit categories on the two places._
+
+Let's presume we already have working information system, and we are able to obtain data we need.
+For the future references we can imagine it as a method with following description
+
+```php
+public function getCategoryDataFromIs(): iterable {
+    // exact implementation is not important
+    // returns an iterable which each element looks like
+    //   [
+    //       'uuid' => '1bb2dd5d-92e0-4263-8ca3-d63aee840c62',
+    //       'parent_uuid' => '3af3c39a-d4b0-4a59-9e36-2e5d54e7ca21',
+    //       'name_en' => 'Electronics',
+    //       'name_cs' => 'Elektronika',
+    //       'description_1' => 'Description of Electronics',
+    //       'description_2' => 'Description of Electronics in Czech',
+    //   ];
+}
+```
+
+Method returns data sorted exactly as they should be on the eshop, meaning category with the same parent will be first, if it comes first from the method.  
+And to make matters worse, IS can only return all 10000 categories.
+
+## First import
+
+!!! note
+    If you're not familiar with the concepts of creating data imports, take a look at the [Basic data import cookbook](./basic-data-import.md)
+
+The main working loop is straightforward
+
+```php
+public function run(): void
+{
+    foreach ($this->getCategoryDataFromIs() as $importData) {
+        $this->importCategory($importData);
+    }
```

Comment:
I am not sure, if this is right thing to do. Because:

1. You have to load all 10 000 categories, but you can load them in batch (I hope) so it would be better practice
2. If it is not possible to batch them - I would maybe start with creating some type of queue (like saving raw data in database), because what will you do, when 9 999 product will fail? 
3. I would prefer to remove importedCategory from source array to better work with memory.

Reply:
Thanks for the ideas, 

1. I believe the exact implementation of the data loader is not important. I declared it as iterable, so you can easily return the generator and don't be bothered about batching here
2. related to 1., and I think it's above the scope of this cookbook
3. related to 1., I don't think it's necessary to have data as an array loaded in memory. Also, it's a good practice to remove elements from the array we currently iterate?

Reply:
Ok, maybe add some note about best practices :)

Reply:
added in 401638197 :)